crates/core/src/pattern/regex.rs

```diff
@@ -115,13 +121,15 @@ impl RegexPattern {
             .filter(|n| n.is_named())
             .map(|n| {
                 VariableCompiler::from_node(
-                    &n,
-                    context,
-                    vars,
-                    vars_array,
-                    scope_index,
-                    global_vars,
-                    logs,
+                    &NodeWithSource::new(n, context.src),
+                    &mut NodeCompilationContext {
+                        compilation: context,
+                        vars,
+                        vars_array,
+                        scope_index,
+                        global_vars,
+                        logs,
+                    },
```

Comment:
Ad-hoc conversions like these will also be cleaned up once the final node compilers are converted.